### DISCUSSION TITLE
About image avatars

### DISCUSSION THREAD

**Discussion by ps96068** [jowilf/starlette-admin#604] [Q&A] [answered]
I'm trying to use starlette-admin in my FastAPI project.
I have a model and this model has the "image_path" field of String type and in which I keep the path to the images saved in the static map of the project.
For this model, I created a ModelView in which I added the "image_path" field.
When I display the data of this model in starlette-admin, the "image_path" value appears, I would like to change it so that the avatar of the image appears.

Can someone guide me with tips on how to do it?

**Comment by jowilf** [comment, marked as answer]
You can create your custom [ImageField](https://jowilf.github.io/starlette-admin/api/fields/#starlette_admin.fields.ImageField) and override the `serialize_value` method. You only need to return a dictionary with `filename`, `content-type`, and `url`. 
    
```py

@dataclass
class MyImageField(BaseImageField):
     async def serialize_value(
        self, request: Request, value: Any, action: RequestAction
    ) -> dict:
              return {"url": request.url_for...}
```

then using it your view

```py
class MyModelView(ModelView):
       fields = [... , MyImageField("avatar"), ... ]
       
```
Example: https://github.com/jowilf/starlette-admin/blob/main/starlette_admin/contrib/sqla/fields.py#L57

**Reply by ps96068** [reply]
Thank you very much for the answer.
With the help of your reply I achieved what I wanted.
With much respect, another question appeared on the same topic:

the HTML code generated for the image avatar is as follows:

```
<td>
  <div class="d-flex">
     <div class="p-1">
     
       <span class="avatar avatar-sm" style="background-image: url(http://127.0.0.1:8000/static/shop/blog1/depositphotos_126693854-stock-photo-set-of-body-care-products .jpg)"></span>
     
     </div>
  </div>
</td>
```

where is this code generated and how can I adapt it to my requirements?


for example, I want to make the following adaptations:
1. to increase the dimensions of the avatar, now it is 2rem
2. add the possibility to open the image in modal when you click on it

**Reply by jowilf** [reply]
you're welcome.

For the listing page follow [this guide](https://jowilf.github.io/starlette-admin/advanced/custom-field/#list-rendering) to provide your custom function. The default one for images [can be found here](https://github.com/jowilf/starlette-admin/blob/bd59d5160fedd5012b8f61d7de936dadcf275fec/starlette_admin/statics/js/render.js#L73).

For the detail page [follow this guide](https://jowilf.github.io/starlette-admin/advanced/custom-field/#detail-page) to provide your own template or override the [default template for images](https://github.com/jowilf/starlette-admin/blob/main/starlette_admin/templates/displays/image.html) in your template directory

**Reply by ps96068** [reply]
Thank you very much for the answer.
